automation/smoke-test-android-components.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+
+# Purpose: Run android-components tests against this application-services working tree.
+# Dependencies: yaml
+# Usage: ./automation/smoke-test-android-components.py
+
+import argparse
+import re
+import subprocess
+import tempfile
+from pathlib import Path
+import yaml
+from shared import step_msg, fatal_err, run_cmd_checked, find_app_services_root, set_gradle_substitution_path
+
+parser = argparse.ArgumentParser(description="Run android-components tests against this application-services working tree.")
+
+group = parser.add_mutually_exclusive_group()
+group.add_argument("--use-local-repo",
+                    metavar="LOCAL_REPO_PATH",
+                    help="Use a local copy of a-c instead of cloning it.")
+group.add_argument("--remote-repo-url",
+                    metavar="REMOTE_REPO_URL",
+                    help="Clone a different a-c repository.")
+parser.add_argument("--branch",
+                    help="Branch of a-c to use.")
+parser.add_argument("--action",
+                    # XXX TODO: it would be very nice to have a "launch sample app" helper here as well.
+                    choices=["run-tests", "do-nothing"],
+                    help="Run the following action once a-c is set up.")
+
+DEFAULT_REMOTE_REPO_URL="https://github.com/mozilla-mobile/android-components.git"
+
+args = parser.parse_args()
+local_repo_path = args.use_local_repo
+remote_repo_url = args.remote_repo_url
+branch = args.branch
+action = args.action
+
+repo_path = local_repo_path
+if repo_path is None:
+    repo_path = tempfile.mkdtemp(suffix="-a-c")
+    if remote_repo_url is None:
+        remote_repo_url = DEFAULT_REMOTE_REPO_URL
+    step_msg(f"Cloning {remote_repo_url}")
+    run_cmd_checked(["git", "clone", remote_repo_url, repo_path])
+    if branch is not None:
+        run_cmd_checked(["git", "checkout", branch], cwd=repo_path)
+elif branch is not None:
+    fatal_err("Cannot specify fenix branch when using a local repo; check it out locally and try again.")
```

Comment:
Would you mind replicating this behavior (no default value for `branch`) in the iOS script for consistency?